CRM/Core/Region.php

```diff
@@ -131,7 +131,7 @@ public function render($default, $allowCmsOverride = TRUE) {
         case 'settings':
           $settingsData = $this->getSettings();
 
-          if (($this->_name === 'html-header') || !CRM_Core_Resources::isAjaxMode()) {
+          if (($this->_name === 'html-header') || (($this->_name !== 'billing-block') && !CRM_Core_Resources::isAjaxMode())) {
```

Comment:
@mattwire I took this change from #18141 along with the unit-test revisions and forward-ported via d279541befb1fe74df614fab2d6549984223430d. I'm not sure the process to reproduce/validate via `r-run`, but the unit test passes.

Tangentially... it's

Reply:
The bug it's supposed to address is when you have more then 1 processor on a page & you switch js doesn't load

Matt had some patches against Omnipay which I merged but then this was logged 
https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/issues/165 and I reverted. I have not looked into it yet but in conjunction with nz.co.fuzion.omnipaymultiprocessor it should work).

So, currently I believe Paypal checkout should load if it is the default processor (but js errors if not). Without the revert the reverse would be true & with the new patch + this it should hopefully work (I have creds if you actually want to try)

Reply:
Thanks @eileenmcnaughton. So in theory I could use that to QA the original #18141 and/or my attempted forward-port (d279541befb1fe74df614fab2d6549984223430d). I tried and can't reproduce what's supposed to be wrong... My report-back is a bit long/detailed, and it's not really about bundles/collections, so I posted on the other PR: https://github.com/civicrm/civicrm-core/pull/18141#issuecomment-679916939

Reply:
Just a bit of follow-up to round-off this subthread - the discussion in 18141 did give a testing procedure, and this forward-port does appear to be working.

Reply:
I've got an alternate proposal about how to render settings here: #18262